markovflow/kernels/sde_kernel.py

```diff
@@ -1032,16 +1165,38 @@ def state_offsets(self, batch_shape: tf.TensorShape, num_transitions: int) -> tf
         We override :meth:`SDEKernel.state_offsets` from the
         parent class to check there is a compatible `batch_shape`.
 
-        :param batch_shape: A tuple of leading dimensions for the state_offsets, where
-            batch_shape can be ``(..., num_kernels)``.
-        :param num_transitions: The number of transitions in the state space model.
-        :return: A tensor of zeros with shape ``batch_shape + [num_transition, state_dim]``,
-            where ``batch_shape = (..., num_kernels)``.
+        :param transition_times: A tensor of times at which to produce matrices, with shape
+            ``batch_shape + [num_transitions]``.
+        :param time_deltas: A tensor of time gaps for which to produce matrices, with shape
+            ``batch_shape + [num_transitions]``.
+        :return: A with shape ``batch_shape + [num_transitions, state_dim]``
         """
+        batch_shape = time_deltas.shape[:-1]
+        num_transitions = tf.shape(time_deltas)[-1]
+
         # raise if batch_shape is provided and batch_shape's last dim is not num_kernel or 1
         self._check_batch_shape_is_compatible(batch_shape)
 
-        return super().state_offsets(batch_shape, num_transitions)
+        result = tf.stack(
+            [
+                augment_matrix(
+                    k.state_offsets(transition_times[..., i, :], time_deltas[..., i, :]),
+                    self.state_dim - k.state_dim,
+                )
+                for i, k in enumerate(self.kernels)
+            ],
+            axis=-3,
+        )
+
+        shape = tf.concat(
+            [
+                batch_shape[:-1],
+                [self.num_kernels, num_transitions, self.state_dim],
+            ],
+            axis=0,
+        )
+        tf.debugging.assert_equal(tf.shape(result), shape)
```

Comment:
`tf.debugging.assert_shapes` for in code checks and documentation is nicer and you don't need to create the concatenation